tracing-appender/src/non_blocking.rs

```diff
@@ -258,7 +263,14 @@ impl Drop for WorkerGuard {
             .sender
             .send_timeout(Msg::Shutdown, Duration::from_millis(100))
         {
-            Ok(_) | Err(SendTimeoutError::Disconnected(_)) => (),
+            Ok(_) => {
+                /* Attempt to wait for `Worker` to flush all messages before dropping. This happens
+                when the `Worker` calls `recv()` on a zero-capacity channel. Use `send_timeout`
+                so that drop is not blocked indefinitely.
+                */
+                let _ = self.shutdown.send_timeout((), Duration::from_millis(1000));
```

Comment:
Why 1000 ms? It might be worth having a comment explaining how this value was chosen. We might want to eventually make this configurable, but that can be done later.

Reply:
I don't have a specific reason for 1000 ms. I just thought this is a reasonable amount of time to give the worker thread to call `recv`.

It seems difficult to determine if there is an ideal value here. Making this configurable would let us pass on the decision of the correct value to the end user. I'll cut a feature request.